modules/omsBidAdapter.js

```diff
@@ -0,0 +1,321 @@
+          ttl: 60,
+          meta: {
+            advertiserDomains: bid?.adomain || []
+          }
+        };
+      });
+    }
+  } catch (e) {
+    logError(e, {id, seatbid});
+  }
+
+  return response;
+}
+
+// Don't do user sync for now
+function getUserSyncs(syncOptions, responses, gdprConsent) {
+  return [];
+}
+
+function onTimeout(timeoutData) {
+  if (timeoutData === null) {
+    return;
+  }
+
+  _trackEvent('timeout', timeoutData);
+}
+
+function onBidderError(errorData) {
+  if (errorData === null || !errorData.bidderRequest) {
+    return;
+  }
+
+  _trackEvent('error', errorData.bidderRequest)
+}
+
+function onBidWon(bid) {
+  if (bid === null) {
+    return;
+  }
+
+  _trackEvent('bidwon', bid)
+}
+
+function _trackEvent(endpoint, data) {
+  ajax(`${TRACK_EVENT_URL}/${endpoint}`, null, JSON.stringify(data), {
+    method: 'POST',
+    withCredentials: false
+  });
+}
+
+function _isMobile() {
+  return (/(ios|ipod|ipad|iphone|android)/i).test(navigator.userAgent);
+}
+
+function _isConnectedTV() {
+  return (/(smart[-]?tv|hbbtv|appletv|googletv|hdmi|netcast\.tv|viera|nettv|roku|\bdtv\b|sonydtv|inettvbrowser|\btv\b)/i).test(navigator.userAgent);
+}
+
+function _getDeviceType() {
+  return _isMobile() ? 1 : _isConnectedTV() ? 3 : 2;
+}
+
+function _getAdMarkup(bid) {
+  let adm = bid.adm;
+  if ('nurl' in bid) {
+    adm += createTrackPixelHtml(bid.nurl);
+  }
+  return adm;
+}
+
+function _isViewabilityMeasurable(element) {
+  return !_isIframe() && element !== null;
+}
+
+function _getViewability(element, topWin, {w, h} = {}) {
+  return getWindowTop().document.visibilityState === 'visible' ? _getPercentInView(element, topWin, {w, h}) : 0;
+}
+
+function _isIframe() {
+  try {
+    return getWindowSelf() !== getWindowTop();
+  } catch (e) {
+    return true;
+  }
+}
+
+function _getMinSize(sizes) {
+  return sizes.reduce((min, size) => size.h * size.w < min.h * min.w ? size : min);
+}
+
+function _getBoundingBox(element, {w, h} = {}) {
+  let {width, height, left, top, right, bottom} = element.getBoundingClientRect();
+
+  if ((width === 0 || height === 0) && w && h) {
+    width = w;
+    height = h;
+    right = left + w;
+    bottom = top + h;
+  }
+
+  return {width, height, left, top, right, bottom};
+}
+
+function _getIntersectionOfRects(rects) {
+  const bbox = {
+    left: rects[0].left, right: rects[0].right, top: rects[0].top, bottom: rects[0].bottom
+  };
+
+  for (let i = 1; i < rects.length; ++i) {
+    bbox.left = Math.max(bbox.left, rects[i].left);
+    bbox.right = Math.min(bbox.right, rects[i].right);
+
+    if (bbox.left >= bbox.right) {
+      return null;
+    }
+
+    bbox.top = Math.max(bbox.top, rects[i].top);
+    bbox.bottom = Math.min(bbox.bottom, rects[i].bottom);
+
+    if (bbox.top >= bbox.bottom) {
+      return null;
+    }
+  }
+
+  bbox.width = bbox.right - bbox.left;
+  bbox.height = bbox.bottom - bbox.top;
+
+  return bbox;
+}
+
+function _getPercentInView(element, topWin, {w, h} = {}) {
```

Comment:
This seems generally useful, please move it to a library and import it

Reply:
Moved to the library